Show bing suggest errors.

kitsunde opened the pull request:
Bing translate returns an error string that doesn't fire the callback when you supply a language that it doesn't recognize (like zh for Chinese), rosetta would display indefinitely show "...", this patch replaces the link text with the error message using the undocumented onerror callback.

If anyone is interested hte Bing API claims to take a [639-1](http://msdn.microsoft.com/en-us/library/ff512399.aspx) languageCode which should be `zh` but that gives an error. Instead it expects `zh-CHS` for simpified and `zh-CHT` for traditional. `zh_HK` doesn't work. This is how I encountered the error.

kitsunde committed:
Show bing suggest errors.

mbi commented:
Thank you for your patch. 

I just tested it in a branch and translation completely stopped working for me. 
It worked again fine when I switched back to the develop branch, so my first impression is a no-go for this PR, hand't had much time to investigate what is wrong, though.

kitsunde commented:
Oh sorry I must've messed something up when I was preparing the commit, ill fix it up when I get back.

On 24 feb 2012, at 19:19, Marco Bonettireply@reply.github.com wrote:

> Thank you for your patch. 
> 
> I just tested it in a branch and translation completely stopped working for me. 
> It worked again fine when I switched back to the develop branch, so my first impression is a no-go for this PR, hand't had much time to investigate what is wrong, though.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/mbi/django-rosetta/pull/26#issuecomment-4155327

kitsunde committed:
Fixes the translation callback not being called and allow for suggestions on "zh-tw" and "zh-cn".

kitsunde commented:
I've fixed the issue. The oncomplete callback was not being called.

I also fixed an issue where "zh-tw" and "zh-cn" (which are standard language codes in Django: https://code.djangoproject.com/browser/django/trunk/django/conf/global_settings.py)  wasn't getting suggestions as the code was only using the first 2 characters of the language code.

mbi commented:
Thank you!
